FirebaseFunctionsSwift/Sources/Codable/Callable+Codable.swift

```diff
@@ -78,116 +50,129 @@ public struct Callable<Request: Encodable, Response: Decodable> {
   }
 
   private let callable: HTTPSCallable
-  private let encoder: StructureEncoder
-  private let decoder: StructureDecoder
+  private let encoder: FirebaseDataEncoder
+  private let decoder: FirebaseDataDecoder
 
-  init(callable: HTTPSCallable, encoder: StructureEncoder, decoder: StructureDecoder) {
+  init(callable: HTTPSCallable, encoder: FirebaseDataEncoder, decoder: FirebaseDataDecoder) {
     self.callable = callable
     self.encoder = encoder
     self.decoder = decoder
   }
 
-  /**
-   * Executes this Callable HTTPS trigger asynchronously.
-   *
-   * The data passed into the trigger must be of the generic `Request` type:
-   *
-   * The request to the Cloud Functions backend made by this method automatically includes a
-   * Firebase Instance ID token to identify the app instance. If a user is logged in with Firebase
-   * Auth, an auth ID token for the user is also automatically included.
-   *
-   * Firebase Instance ID sends data to the Firebase backend periodically to collect information
-   * regarding the app instance. To stop this, see `[FIRInstanceID deleteIDWithHandler:]`. It
-   * resumes with a new Instance ID the next time you call this method.
-   *
-   * - Parameter data: Parameters to pass to the trigger.
-   * - Parameter completion: The block to call when the HTTPS request has completed.
-   *
-   * - Throws: An error if any value throws an error during encoding.
-   */
+  /// Executes this Callable HTTPS trigger asynchronously.
+  ///
+  /// The data passed into the trigger must be of the generic `Request` type:
+  ///
+  /// The request to the Cloud Functions backend made by this method automatically includes a
+  /// FCM token to identify the app instance. If a user is logged in with Firebase
+  /// Auth, an auth ID token for the user is also automatically included.
+  ///
+  /// Firebase Cloud Messaging sends data to the Firebase backend periodically to collect information
+  /// regarding the app instance. To stop this, see `Messaging.deleteData()`. It
+  /// resumes with a new FCM Token the next time you call this method.
+  ///
+  /// - Parameter data: Parameters to pass to the trigger.
+  /// - Parameter completion: The block to call when the HTTPS request has completed.
+  ///
+  /// - Throws: An error if any value throws an error during encoding.
   public func call(_ data: Request,
                    completion: @escaping (Result<Response, Error>)
-                     -> Void) throws {
-    let encoded = try encoder.encode(data)
+                     -> Void) {
```

Comment:
This isn't marked as throws anymore - that makes sense given the implementation, right? If so, we need to update the doc